game-core/src/main/java/org/triplea/debug/ErrorMessage.java

```diff
@@ -130,7 +132,12 @@ private void setUploadRecord(final LogRecord logRecord) {
   }
 
   private void activateUploadErrorReportButton(final URI lobbyUri, final LogRecord logRecord) {
-    INSTANCE.uploadButton.setVisible(true);
+    // Set upload button to be visible only if the logging is greater than a 'warning'.
+    // Warning logging should tell a user how to correct the error and it should be something
+    // that is 'normal' (but maybe not happy, like no internet) and something they can fix
+    // (or perhaps something that we simply can never fix). Hence, for warning we want to inform
+    // the user that the problem happened, how to fix it, and let them create a bug report manually.
+    INSTANCE.uploadButton.setVisible(logRecord.getLevel().intValue() > Level.WARNING.intValue());
```

Comment:
I actually think this problem is too complex to represent it in a single severity level.
Let's say a map maker is trying to build a new map and wants to test it.
Turns out they got the format wrong and therefore the engine is unable to parse the map. I would still consider this a severe error, not a warning because it is an exception that prevents an entire action from happening, but it shouldn't be reportable.
However let's say $User has their maps saved as a folder somewhere and the write and read permissions are messed up. In this case the error is still severe and it's a clear case for user help, but the system can't communicate that at all.

I must admit that I don't have a better idea than this, it's probably the best short-term approach we can go with, but we might want to change the system to do some heuristics if the error is user related (local IOException, FileNotFound etc.) or it has a real code implication (NullPointerException etc.) in the long run.
Of course there are edgecases as well: Is a StackOverflowError a regression or a low-memory user issue?
But I think it's better than this approach

Reply:
@RoiEXLab see my response to the PR, but secondarily I'd suggest warning is still okay. The game did not crash. IMO `Severe` is the game is left in a dead state, warning is basically an action is not going to be available. INFO probably is not really worth using (pointless). I really can't emphasize enough how the logging perspective for a thick client is different compared to say a web server.